### PR TITLE
Add common/base GUI code and GUI to mark the damaged area (solar panels)

### PR DESCRIPTION
**Pull request by xiejxie**

### CODE REVIEW

**Comment by Timothylock** [comment]
I'm missing a bunch of dependencies. Can you please include what is needed in your readme and either generate a requirements package or add pip install scripts into the `/setup` directory?

File: src/task1/mark-damaged-module/markergui.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        self.dots[filename] = new_arr
+        print(x, y)
+        self.next_frame(self.index)
+
+if __name__ == "__main__":
+    dir_path = '../common/test/'
+    filelist = [dir_path + x for x in os.listdir(dir_path)]
+    root = Tk()
+    app = MarkerViewer(root, filelist, 1000, 500)
+    root.mainloop()
```
Comment (Timothylock) [inline src/task1/mark-damaged-module/markergui.py:61]: Not sure where to put this comment, but it should also take in another sys arg to output a json file :D 

```

Output:
A JSON file marking coordinates and filename of broken panels

{"damaged": [
  {"long": 123.456,
   "lat": 43.1234,
   "filename": "img_0123.jpg"
  },
{"long": 125.456,
   "lat": 46.1234,
   "filename": "img_0124.jpg"
  }
]
}
```

(Also looks like the program never ends and keeps relooping through all the images ;_;

File: src/task1/common/gui.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+from tkinter import Label, Frame, Button, Entry, StringVar
```
Comment (Timothylock) [inline src/task1/common/gui.py:1]: THANK YOU 💯💯💯💯

File: src/task1/mark-damaged-module/markergui.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                print('got em')
+                new_arr.append(coords)
+        if len(new_arr) == len(self.dots[filename]):
+            new_arr.append([x, y])
+        self.dots[filename] = new_arr
+        print(x, y)
+        self.next_frame(self.index)
+
+if __name__ == "__main__":
+    dir_path = '../common/test/'
```
Comment (Timothylock) [inline src/task1/mark-damaged-module/markergui.py]: Remember 

```
Input:
A JSON file marking which files definitely contain broken solar panels

{"positive": ["img_0123.jpg", "img_0124.jpg"]}
```

refer to the ticket for more info :)

File: src/task1/mark-damaged-module/markergui.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        filename = self.files[self.index]
+        new_arr = []
+        for coords in self.dots[filename]:
+            if x not in range(coords[0]-10,coords[0]+11) or y not in range(coords[1]-10,coords[1]+11):
+                print('got em')
+                new_arr.append(coords)
+        if len(new_arr) == len(self.dots[filename]):
+            new_arr.append([x, y])
+        self.dots[filename] = new_arr
+        print(x, y)
```
Comment (Timothylock) [inline src/task1/mark-damaged-module/markergui.py]: You'll eventually need to call Lev's projection code so I would recommend opening up the metadata file, etc etc and getting the data ready to call that function in the future. 

https://github.com/UTAT-AeRo/vision_201819/pull/1#pullrequestreview-180569314

If you need sample images: https://data.timothylock.ca/index.php/s/wHdd6B3EWFjXAot

Thankssssss

**Review by Timothylock — CHANGES_REQUESTED** [review]
I don't like how it puts a red X.... Thanks so much for bringing common code out. Just a few things I noticed

File: src/task1/mark-damaged-module/markergui.py (hunk 3 above)
Comment (Timothylock) [reply src/task1/mark-damaged-module/markergui.py]: So it should take in a filepath to a json file

File: src/task1/mark-damaged-module/markergui.py (hunk 1 above)
Comment (xiejxie) [reply src/task1/mark-damaged-module/markergui.py:61]: Oh I'm not done with this yet xD Sorry, should have specified that.
Would you rather we only make incremental updates to our branch and then push when it's all done? I thought it'd be safer to push WIP so we can checkpoint somewhere.

Thanks for detailed review btw!

Comment (Timothylock) [reply src/task1/mark-damaged-module/markergui.py:61]: Oh okay that's fine. I love incremental. Next time you assign me, just tell me what you have left so I know to keep that in mind when reviewing :)

**Review by Timothylock — APPROVED** [review]

**Pull request merged**